src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SymfonyRuntime implements RuntimeInterface
+{
+    private $request;
+    private $input;
+    private $output;
+    private $application;
+    private $command;
+
+    public function __construct(array $options = [])
+    {
+        if (isset($_SERVER['argv']) && class_exists(ArgvInput::class)) {
```

Comment:
Can you please explain the purpose of this clause? IIUC later code can call create getInput on it's own it it needs it. And moreover - it does not the check `class_exists(ArgvInput::class)` at that time. So looks like if `ArgvInput` class does not exist - code may try to load unexisting class at some point later here
https://github.com/symfony/symfony/blob/06d70e0190b9f31b7a3416ce94a5086fa76f850e/src/Symfony/Component/Runtime/SymfonyRuntime.php#L193

Isn't this clause redundand?

Reply:
The input must be parsed before `.env` files are loaded.

Reply:
In the other places where `getInput()` is called, the class is guaranteed to always exists.

Reply:
Got it. Very "know-how" part here. Thank you for explanation